extensions/amp-analytics/0.1/events.js

```diff
@@ -447,21 +477,144 @@ export class TimerEventTracker extends EventTracker {
         'Bad maxTimerLength specification');
     const callImmediate = 'immediate' in timerSpec ?
         Boolean(timerSpec['immediate']) : true;
+    const timerStart = 'startSpec' in timerSpec ? timerSpec['startSpec'] : null;
+    user().assert(!timerStart || typeof timerStart == 'object',
+        'Bad timer start specification');
+    const timerStop = 'stopSpec' in timerSpec ? timerSpec['stopSpec'] : null;
+    user().assert((!timerStart && !timerStop) || typeof timerStop == 'object',
+        'Bad timer stop specification');
+
+    const timerId = this.generateTimerId_();
+    const timerParams = {};
+    timerParams[this.TIMER_PARAMS_.INTERVAL_LENGTH] = interval;
+    timerParams[this.TIMER_PARAMS_.MAX_TIMER_LENGTH] = maxTimerLength;
+    timerParams[this.TIMER_PARAMS_.CALL_IMMEDIATE] = callImmediate;
+    this.trackers_[timerId] = timerParams;
+    if (!timerStart) {
+      this.startTimer_(timerId, eventType, listener);
+      this.trackers_[timerId][this.TIMER_PARAMS_.CAN_RESTART] = false;
+    } else {
+      this.trackers_[timerId][this.TIMER_PARAMS_.CAN_RESTART] = true;
+      const startTracker = trackerProvider(timerStart);
```

Comment:
is it possible that we don't pass the `trackerProvider` here but create a new tracker based on the `timerStart`/`timerStop` spec? 
We've done [similar things](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/events.js#L676) to visibilityTracker `waitFor` spec. 
This would first makes the code looks cleaner in my opinion, and provide flexibility to restrict the type of eventTracker that a timerTracker can create. For example maybe we should restrict the recursive use case like
```
on : timer,
startSpect: {
  on: timer,
}
```

Regarding why we only create one tracker for each event type, I believe that's because we can then dispose them properly in analytics root. But as long as timerTracker can keep track of all the other event tracker it creates and dispose them correctly, I think we should be good.   
I remembered we agreed the cost of creating an eventTracker is minimum right? @dvoytenko